### PR TITLE
Haskell Compatibility ported from F#+

### PR DESCRIPTION
**Pull request by gusty**
This is all the code we had in FSharpPlus for Haskell Compatibility, including sample script.
The dependency to F#+ is at version 1.0.0-CI00089 which is the latest in .NET 4.0.
Once this project is migrated to 4.5 or .NET standard we can move to latest F#+ version, with minor adjustments.

### CODE REVIEW

**Comment by jmquigs** [comment]
Nice work!  My knowledge of Haskell is quite limited, but I was able to build and run the sample script.    This LGTM, I'll leave the PR open for a day or so in case anyone else has comments.

I will also make an effort to upgrade this project to a more recent F#.

File: FSharp.Compatibility.Haskell/Prelude.fs [hunk 1]
```diff
@@ -1,55 +1,344 @@
...
+| produced by the       |                       |                    |
+| first as an argument  |                       |                    |
+| to the second.        |                       |                    |
++-----------------------+-----------------------+--------------------+
+| Bind with             |                       |                    |
+| interchanged          | =<<                   | =<<                |
+| arguments.            |                       |                    |
++-----------------------+-----------------------+--------------------+
+```
+*)
```
Comment (wallymathieu) [inline FSharp.Compatibility.Haskell/Prelude.fs]: Nice!

File: FSharp.Compatibility.Haskell/Sample.fsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+let askString next = do' {
+  do! (liftIO $ putStrLn "Please enter a string") 
+  let! s = liftIO $ getLine
+  return! next s}
+
+// reportResult :: String -> IO ()
+let reportResult s = do' {
+  return! putStrLn ("You entered: " + s) }
+  
+let mainAction = runContT (callCC askString) reportResult       //try -> runIO mainAction
```
Comment (wallymathieu) [inline FSharp.Compatibility.Haskell/Sample.fsx]: Makes this library much more useful 👍

**Comment by gusty** [comment]
Thanks @jmquigs if I have time I will add more Haskell stuff to this project. And let me know if you upgrade the Framework Version and need to update the F#+ version.

**Pull request merged**